### ISSUE TITLE
[BUG|clang-tidy] false positive on misc-unconventional-assign-operator

### ISSUE THREAD

**Issue by yaoxinliu** [llvm/llvm-project#143237]
```
struct A {
    explicit A(int) {
    }

    A& operator=(int n) {
        // ok, no clang-tidy warning.
        return *this = A(n); 
    }
};

template<typename T>
struct B {
    explicit B(int) {
    }

    B& operator=(int n) {
        // clang-tidy warning: Operator=() should always return '*this'
        // clang-tidy    rule: misc-unconventional-assign-operator
        return *this = B(n); 
    }
};
```

clang-tidy version: 20.1.5
